src/hoc/withMousePosition.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+const withMousePosition = (WrappedComponent) => {
+  class Hoc extends React.Component {
+    state = {
+      x: 0,
+      y: 0
+    };
+
+    componentDidMount() {
+      document.body.addEventListener(
+          'mousemove', 
+          this.updateMousePostion.bind(this)
+        );
+    }
+
+    updateMousePostion(e) {
+      this.setState({
+        x: e.pageX,
+        y: e.pageY
+      });
+    }

```

Comment:
Try to change this function to an arrow function so the .bind(this) could be removed.
In my current company, we have an approach that we use only arrow functions. You can also try to keep that convention.